### PR TITLE
Add client tests

### PR DESCRIPTION
**Pull request by deribaucourt**

### CODE REVIEW

File: __mocks__/vscode.ts [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+      fire: jest.fn(),
+    };
+  });
+
+  const QuickPickItemKind = {
+    Separator: -1,
+    Default: 0,
+  };
+
+  export = {
```
Comment (idillon-sfl) [inline __mocks__/vscode.ts:108]: On my side typescript complains it is using the CommonJS syntax, while the EcmaScript syntaxt should be used (`export default`). Yet it won't compile if I make that change.

I am not sure where is exactly the problem, but I noticed .eslintrc.js is also using a commonjs export. I also noticed you put this file into "ignorePatterns". I am thinking this was an attempt to solve that problem.

Comment (deribaucourt) [reply __mocks__/vscode.ts:108]: This file comes from another repo and works fine. As long as we don't change it, it makes no sense to me to test it's linting, which follows a different standard in another project. Also it makes it easy to update this file in the future when the vscode API changes/adds new features. I suggest to ignore linting this file and keep it as original. A comment at the top clearly indicates where it comes from.

File: __mocks__/vscode.ts [hunk 2]
```diff
@@ -0,0 +1,134 @@
+// Jtest Mock from https://github.com/jest-community/vscode-jest/blob/1867bde9ebf8c21d45409e7ad8a95a743ac4390d/__mocks__/vscode.ts
+// under MIT License
+
+const languages = {
+    createDiagnosticCollection: jest.fn(),
```
[the remark this answers is not in the document]
Comment (deribaucourt) [reply __mocks__/vscode.ts:5]: As discussed in another comment, I suggest keeping the original file and not changing it's linting rules.

File: jest.config.js [hunk 3]
```diff
@@ -21,11 +21,11 @@ module.exports = {
     ]
   },
   testMatch: [
-    "<rootDir>/**/__tests__/*.ts"
+    "<rootDir>/**/__tests__/**/*.ts"
   ],
   collectCoverageFrom: [
     "**/*.ts",
-    "!**/__test__/*",
+    "!**/__test__/**/*.ts",
```
[the remark this answers is not in the document]
Comment (deribaucourt) [reply jest.config.js:28]: Done, but testMatch: was the relevant field.

File: __mocks__/vscode.ts (hunk 1 above)
Comment (idillon-sfl) [reply __mocks__/vscode.ts:108]: Makes sense. However I am surprised it works. Both syntaxes are not supposed to be interchangeable within a project. As a proof, this file does not compile if you use the ecmascript syntax. It makes me think the "root project" is not configured properly.

**Review by idillon-sfl — APPROVED** [review]

File: server/src/server.ts [hunk 4]
```diff
@@ -33,8 +34,8 @@ const bitBakeDocScanner = new BitBakeDocScanner()
 const bitBakeProjectScanner: BitBakeProjectScanner = new BitBakeProjectScanner(connection)
 const contextHandler: ContextHandler = new ContextHandler(bitBakeProjectScanner)
 
-connection.onInitialize(async (params): Promise<InitializeResult> => {
-  const workspaceRoot = params.rootPath ?? ''
+connection.onInitialize(async (params: InitializeParams): Promise<InitializeResult> => {
+  const workspaceRoot = params.workspaceFolders?.[0]?.uri ?? ''
```
Comment (WilsonZiweiWang) [inline server/src/server.ts:38]: I think it is possible that users have multiple folders opened in one workspace and all of them are bitbake projects.

**Pull request merged**